internals/pasture/CMakeLists.txt

```diff
@@ -0,0 +1,14 @@
+if(ENABLE_psi4pasture)
+  ExternalProject_Add(psi4pasture
+    DEPENDS psi4_core
+    GIT_REPOSITORY https://github.com/psi4/psi4pasture
+    CMAKE_ARGS -DENABLE_ccsort=${ENABLE_ccsort}
+               -DENABLE_transqt2=${ENABLE_transqt2}
+               -Dpsi4_DIR=${STAGED_INSTALL_PREFIX}/share/cmake/psi4
+               -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
+               -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}
+               -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}
+    CMAKE_CACHE_ARGS -DCMAKE_C_FLAGS:STRING=${CMAKE_C_FLAGS}
+                     -DCMAKE_CXX_FLAGS:STRING=${CMAKE_CXX_FLAGS}
+    INSTALL_COMMAND ${CMAKE_MAKE_PROGRAM} install)
```

Comment:
The pasture build systems uses psi4's config file to locate the driver directory and everything gets installed within `psi4_install_prefix/lib/psi4/driver/pasture`. 

I thought this would be a safe way to make sure that a pasture install is local to a *particular* psi4 install. If pasture gets installed into`PYTHONPATH` then any psi4 can find it, but there would probably be strange errors if the compiler-version, math libs, ect. didn't match between the psi4 and the pasture builds. 


We don't have an install procedure for plugins, but I guess I have artificially made one for the "pasture" plugins. The pasture `CMakeLists.txt` is not a clone of the one we give with plugins, but It is heavily inspired by it (since it was my starting point). I am trying to use as much info as possible from the psi4 config file, so that the build env for pasture is as close as possible to the "host" psi4. 

The way things are currently set up, there is no mechanism for detecting a pre-built pasture, rather it will always be built if requested. If you are building pasture outside of psi4's build system, then it will have to be built after, and will require a valid `psi4_DIR` be passed as an argument to cmake. The equivalent using `CMAKE_PREFIX_PATH` would work as well.